07_iterative_algorithms.md

```diff
@@ -69,6 +69,82 @@ Similarly to creating an `union` it is possible to `join` multiple RDDs at once
 
 #### Pushing Transformations to the Local Data Structures
 
+Relatively simple but powerful and efficient way to reduce the length of the lineage is to replace consecutive narrow transformations with their local equivalent. Since basic methods from RDD API have identical equivalents in `Iterator` API it is as simple as using `mapPartitions`. For example following chain of transformations:
```

Comment:
Point is there is minimal "intellectual overhead". We use similar API composed from higher order functions and function we pass to these have exactly the same signatures. So the same was going from local to distributed in Spark we can move in the opposite direction.